test/byContract/IexecEscrow/IexecEscrowNative.ts

```diff
@@ -0,0 +1,302 @@
+                            nativeAmountSent.sub(excessAmountWei),
+                        ],
+                    )
+                    .to.emit(iexecPocoAsAnyone, 'Transfer')
+                    .withArgs(ZERO_ADDRESS, iexecAdmin.address, amounts[0]);
+
+                await checkBalances(iexecAdmin.address, amounts[0]);
+                expect(await ethers.provider.getBalance(iexecAdmin.address)).to.equal(
+                    initialBalance.sub(nativeAmountSent).add(excessAmountWei),
+                );
+                await checkBalances(anyone.address, amounts[1]);
+            });
+
+            it('Should not depositForArray with mismatched array lengths', async () => {
+                const amounts = [DEPOSIT_AMOUNT.mul(2), DEPOSIT_AMOUNT, DEPOSIT_AMOUNT.div(2)];
+                const totalAmount = amounts.reduce((a, b) => a.add(b), BigNumber.from(0));
+                const nativeAmountSent = ethers.utils.parseUnits(totalAmount.toString(), 9);
+                const targets = [iexecAdmin.address, anyone.address];
+
+                await checkBalances(iexecAdmin.address, BigNumber.from(0));
+                await checkBalances(anyone.address, BigNumber.from(0));
+
+                await expect(
+                    iexecPoco.depositForArray(amounts, targets, { value: nativeAmountSent }),
+                ).to.be.revertedWith('invalid-array-length');
+            });
+        });
+
+        describe('Withdrawals', () => {
+            it('Should withdraw native tokens', async () => {
+                await iexecPocoAsAnyone.deposit({ value: DEPOSIT_AMOUNT_ETHER });
+
+                expect(await iexecPocoAsAnyone.callStatic.withdraw(DEPOSIT_AMOUNT)).to.be.true;
+                await expect(iexecPocoAsAnyone.withdraw(DEPOSIT_AMOUNT))
+                    .to.changeEtherBalances(
+                        [anyone, iexecPoco],
+                        [DEPOSIT_AMOUNT_ETHER, -DEPOSIT_AMOUNT_ETHER],
+                    )
+                    .to.emit(iexecPocoAsAnyone, 'Transfer')
+                    .withArgs(anyone.address, ZERO_ADDRESS, DEPOSIT_AMOUNT);
+
+                await checkBalances(anyone.address, BigNumber.from(0));
+            });
+
+            it('Should withdraw native tokens to another address', async () => {
+                await expect(
+                    iexecPoco.deposit({ value: DEPOSIT_AMOUNT_ETHER }),
+                ).to.changeEtherBalances(
+                    [iexecAdmin, iexecPoco],
+                    [-DEPOSIT_AMOUNT_ETHER, DEPOSIT_AMOUNT_ETHER],
+                );
+                expect(await iexecPocoAsAnyone.balanceOf(iexecAdmin.address)).to.equal(
+                    DEPOSIT_AMOUNT,
+                );
+
+                const withdrawToArgs = [DEPOSIT_AMOUNT, anyone.address] as [BigNumber, string];
+                expect(await iexecPoco.callStatic.withdrawTo(...withdrawToArgs)).to.be.true;
+                await expect(iexecPoco.withdrawTo(...withdrawToArgs))
```

Comment:
Maybe you can use same `...args` pattern everywhere

Reply:
Done